src/utils/index.ts

```diff
@@ -152,24 +145,34 @@ export function formatData(data: Data, args: Inputs): Output {
     // Pass any required query params with user values for relevant scripts
     scripts: data.scripts
       ? data.scripts.map((script) => {
+          const paramKeys = script.params;
+          const paramInputs = filterArgs(args, paramKeys);
+          const optionalParamKeys = script.optionalParams
+            ? Object.keys(script.optionalParams)
+            : [];
+
+          // if we're receiving undefined or null as an ar, we check the optionalParams
+          const newArgs: any = {};
+          Object.keys(args).forEach((key) => {
+            newArgs[key] = args[key] || script.optionalParams?.[key];
+          });
+          const optionalParamInputs = filterArgs(newArgs, optionalParamKeys);
```

Comment:
newArgs is not pretty, if there's a fancy way, please let me know... and I'll change

Reply:
We could use a reducer if you thikn it's not pretty (but it may end up even more uglier :joy:)